sdks/python/apache_beam/runners/portability/expansion_service.py

```diff
@@ -41,7 +43,18 @@ def __init__(self, options=None):
 
   def Expand(self, request, context=None):
     try:
-      pipeline = beam_pipeline.Pipeline(options=self._options)
+      options = copy.deep_copy(self._options)
+      request_options = pipeline_options.PipelineOptions.from_runner_api(
+          request.pipeline_options)
+      # TODO(https://github.com/apache/beam/issues/20090): Figure out the
+      # correct subset of options to apply to expansion.
+      if request_options.view_as(
```

Comment:
same comment re: being not just for GCP

Reply:
I put it here as it's next to the other update options. Open to suggestions if there's a better place. (A new update options?)

Reply:
Fixed.